pkg/performanceprofile/controller/performanceprofile_controller.go

```diff
@@ -417,22 +434,27 @@ func (r *PerformanceProfileReconciler) Reconcile(ctx context.Context, req ctrl.R
 
 		// remove finalizer
 		if hasFinalizer(instance, finalizer) {
+			klog.InfoS("Remove finalizer", "instance", instanceKey.String())
 			removeFinalizer(instance, finalizer)
-			if err := r.Update(ctx, instance); err != nil {
+			if err := r.ManagementClient.Update(ctx, instance); err != nil {
 				return reconcile.Result{}, err
 			}
+			klog.InfoS("Finalizer Removed", "instance", instanceKey.String())
 			return reconcile.Result{}, nil
 		}
 	}
 
 	// add finalizer
 	if !hasFinalizer(instance, finalizer) {
+		klog.InfoS("Add finalizer", "instance", instanceKey.String())
 		instance.SetFinalizers(append(instance.GetFinalizers(), finalizer))
-		instance.Status.Conditions = status.GetProgressingConditions("DeploymentStarting", "Deployment is starting")
-		if err := r.Update(ctx, instance); err != nil {
+		if prof, ok := instance.(*performancev2.PerformanceProfile); ok {
+			prof.Status.Conditions = status.GetProgressingConditions("DeploymentStarting", "Deployment is starting")
```

Comment:
We should use the `r.StatusWriter.Update` to update conditions instead

Reply:
Done. Thnkas

Reply:
@rbaturov My bad, we are calling Update two lines below, and we cannot call update twice for the same object (since the resource version is changing) for hypershift flow it won't be a problem because the status resides on separate object.
We'll have to figure out the implementation when time comes.